src/js/other-websites/inpage_menu.js

```diff
@@ -51,226 +52,610 @@ async function isUserSignedIn() {
-  await setCustomFonts();
+  if (!serverStoragePrefInStorage) {
+    return await browser.runtime.sendMessage({
+      method: "getServerStoragePref",
+    });
+  } else {
+    return serverStoragePrefInStorage;
+  }
+}
 
-  const signedInUser = await isUserSignedIn();
-  const signedOutContent = document.querySelector(".fx-content-signed-out");
-  const signedInContent = document.querySelector(".fx-content-signed-in");
+async function getMasks(options = { subdomainSet: false }) {
+  const serverStoragePref = await getServerStoragePref();
 
-  document.addEventListener("keydown", handleKeydownEvents);
+  if (serverStoragePref) {
+    try {
+      const masks = await browser.runtime.sendMessage({
+        method: "getAliasesFromServer",
+        options,
+      });
 
-  if (!signedInUser) {
-    signedOutContent.classList.remove("is-hidden");
-    // Remove signed in content from DOM so there are no hidden/screen readable-elements available
-    signedInContent.remove();
+      return masks;
+    } catch (error) {
+      console.warn(`getAliasesFromServer Error: ${error}`);
 
-    const signUpMessageEl = document.querySelector(
-      ".fx-relay-menu-sign-up-message"
-    );
+      // API Error — Fallback to local storage
+      const { relayAddresses } = await browser.storage.local.get(
+        "relayAddresses"
+      );
 
-    signUpMessageEl.textContent = browser.i18n.getMessage(
-      "pageInputIconSignUpText"
-    );
+      return relayAddresses;
+    }
+  }
 
-    const signUpButton = document.querySelector(
-      ".fx-relay-menu-sign-up-btn"
-    );
+  // User is not syncing with the server. Use local storage.
+  const { relayAddresses } = await browser.storage.local.get("relayAddresses");
+  return relayAddresses;
+}
 
-    signUpButton.textContent = browser.i18n.getMessage(
-      "pageInputIconSignUpButton"
-    );
+async function fillTargetWithRelayAddress(generateClickEvt) {
+  sendInPageEvent("click", "input-menu-reuse-alias");
+  preventDefaultBehavior(generateClickEvt);
 
-    signUpButton.addEventListener("click", async (clickEvt) => {
-      preventDefaultBehavior(clickEvt);
-      await browser.runtime.sendMessage({
-        method: "openRelayHomepage",
-      });
-      sendInPageEvent("click", "input-menu-sign-up-btn");
-      iframeCloseRelayInPageMenu();
-    });
+  const maskAddress = generateClickEvt.target.dataset.mask;
 
-    sendInPageEvent("viewed-menu", "unauthenticated-user-input-menu");
+  await browser.runtime.sendMessage({
+    method: "fillInputWithAlias",
+    message: {
+      filter: "fillInputWithAlias",
+      newRelayAddressResponse: {
+        address: maskAddress,
+      },
+    },
+  });
+}
 
-    // Focus on "Go to Firefox Relay" button
-    signUpButton.focus();
+async function populateMaskList(
+  maskList,
+  masks,
+  options = { replaceMaskAddressWithLabel: false }
+) {
+  const list = maskList.querySelector("ul");
 
+  if (masks.length === 0) {
+    maskList.remove();
```

Comment:
I refactored this to just `return` at this stage, and rewrote the CSS to hidden by default. When there are masks, it will add a `is-visible` class.

I realized I was doing something similiar to the search bar (removing it instead of just hiding it): https://github.com/mozilla/fx-private-relay-add-on/blob/mpp-830-sign-back-in-page/src/js/other-websites/inpage_menu.js#L366-L372 

(Search bar refactored in 6e56c2be78e42db03e6abdbd5075c3d17e03b888)